merge.go

```diff
@@ -0,0 +1,659 @@
+			op := updateEdge{
+				e:     d.edge,
+				id:    d.id,
+				typ:   d.addLabel,
+				props: d.setProp,
+			}
+			res = append(res, op.writeQuery(cfg))
+		case createEdgeOp:
+			op := createEdge{
+				e:     d.edge,
+				id:    d.id,
+				typ:   d.addLabel,
+				props: d.setProp,
+			}
+			res = append(res, op.writeQuery(cfg))
+		}
+	}
+	return res, vars
+}
+
+type OperationQueue struct {
+	Ops  []string
+	vars []map[string]interface{}
+}
+
+func RunOperations(ctx *ls.Context, session *Session, tx neo4j.Transaction, ops OperationQueue) error {
+	for ix, op := range ops.Ops {
+		fmt.Println(op)
+		_, err := tx.Run(op, ops.vars[ix])
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func Merge(memGraph, dbGraph *lpg.Graph, dbGraphIds map[*lpg.Node]int64, dbEdges map[*lpg.Edge]int64, config Config) (*lpg.Graph, OperationQueue, error) {
+	memEntitiesMap := ls.GetEntityInfo(memGraph)
+	dbEntitiesMap := ls.GetEntityInfo(dbGraph)
+	nodeAssocations := make(map[*lpg.Node][]step)
+	edgeAssocations := make(map[*lpg.Edge][]edgeStep)
+
+	findDBCounterpart := func(n *lpg.Node, schemaPV, idPV *ls.PropertyValue) *lpg.Node {
+		if ls.AsPropertyValue(n.GetProperty(ls.EntitySchemaTerm)).IsEqual(schemaPV) {
+			if ls.AsPropertyValue(n.GetProperty(ls.EntityIDTerm)).IsEqual(idPV) {
+				return n
+			}
+		}
+		return nil
+	}
+	memEntities := make([]*lpg.Node, len(memEntitiesMap))
+	dbEntities := make([]*lpg.Node, len(memEntitiesMap))
+	// align by index matching entity roots i.e, first entity in mem slice should match first entity in db slice
+	ix := 0
+	for n := range memEntitiesMap {
+		schemaPV := ls.AsPropertyValue(n.GetProperty(ls.EntitySchemaTerm))
+		eidPV := ls.AsPropertyValue(n.GetProperty(ls.EntityIDTerm))
+		memEntities[ix] = n
+		for db := range dbEntitiesMap {
+			if findDBCounterpart(db, schemaPV, eidPV) != nil {
+				dbEntities[ix] = db
+				delete(dbEntitiesMap, db)
+				continue
+			}
+		}
+		ix++
+	}
+	deltas := make([]delta, 0)
+	for ix := 0; ix < len(memEntities); ix++ {
+		d, _ := mergeEntity(memEntities[ix], dbEntities[ix], dbGraph, dbGraphIds, dbEdges, nodeAssocations, edgeAssocations)
+		deltas = append(deltas, d...)
+	}
+	// for _, steps := range nodeAssocations {
+	// 	for _, step := range steps {
+	// 		for edgeItr := memGraph.GetEdges(); edgeItr.Next(); {
+	// 			edge := edgeItr.Edge()
+	// 			if ls.GetEntityRootNode(edge.GetFrom()) != ls.GetEntityRootNode(edge.GetTo()) {
+	// 				if _, ok := dbEdges[edge]; !ok {
+	// 					dbFrom := dbGraph.NewNode(edge.GetFrom().GetLabels().Slice(), ls.CloneProperties(edge.GetFrom()))
+	// 					dbEdge := dbGraph.NewEdge(dbFrom, step.node, edge.GetLabel(), ls.CloneProperties(edge))
+	// 					de := compareGraphEdge(edge, dbEdge, mergeOp)
+	// 					de.id = dbEdges[dbEdge]
+	// 					de.operation = updateEdgeOp
+	// 					de.setDeltaToEdge(dbEdge)
+	// 					deltas = append(deltas, de)
+	// 				}
+	// 			}
+	// 		}
+	// 	}
+	// }
+	queries, vars := buildQueriesFromDeltas(deltas, config)
```

Comment:
Move this out to a separate func that will be called to print out the operations
Write another func  runQueries([]Delta,config)
This should creates the nodes first, get their ids, put into node id map (call CreateNodes func),  link the edges (delta should point to the from/to dbnodes, so you can lookup their ids in the nodeIdMap), and then create/update edges
linking edges: If you have a createEdge delta with nodes n1, n2, lookup the nodes in the node id map, and write the createEdge query using those node ids.